pkg/install/deploystorage.go

```diff
@@ -324,12 +324,20 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 			return err
 		}
 
+		// Sometimes we get into the race condition between external services modifying
+		// subnets and our validation code. We try to catch this early, but
+		// these errors is propagated to make the user-facing error more clear incase
+		// modification happened after we ran validation code and we lost the race
 		if s.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
 			if strings.EqualFold(*s.SubnetPropertiesFormat.NetworkSecurityGroup.ID, nsgID) {
 				continue
 			}
 
-			return fmt.Errorf("tried to overwrite non-nil network security group")
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided subnet '%s' is invalid: must not have a network security group attached.", subnetID)
+		}
+
+		if !strings.EqualFold(*s.PrivateLinkServiceNetworkPolicies, "Disabled") {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided VM subnet '%s' is invalid: must have privateLinkServiceNetworkPolicies disabled.", subnetID)
```

Comment:
Do you have a specific reason to add this additional check?  There's still a race because we're not yet at the point where we are adding the PLS to the subnet.  The setting could still be changed between now and then, and presumably if that happens the second deploy will fail?

Reply:
Will open separate PR for this